### PR TITLE
added gda

### PR DESCRIPTION
**Pull request by ametel01**

### CODE REVIEW

File: crates/dojo-defi/src/dutch_auction/gda.cairo [hunk 1]
```diff
@@ -0,0 +1,43 @@
+use cubit::types::fixed::{Fixed, FixedType, FixedMul, FixedSub, FixedDiv};
+
+#[derive(Component)]
+#[derive(Drop)]
```
Comment (tarrencev) [inline crates/dojo-defi/src/dutch_auction/gda.cairo:4]:
```suggestion
#[derive(Component, Drop)]
```

These can be combined

File: crates/dojo-defi/src/dutch_auction/gda.cairo [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+struct Market {
```
Comment (tarrencev) [inline crates/dojo-defi/src/dutch_auction/gda.cairo:5]:
```suggestion
struct GradualDutchAuction {
```

File: crates/dojo-defi/src/dutch_auction/gda.cairo [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    initial_price: FixedType,
+    scale_factor: FixedType,
+    decay_constant: FixedType,
+    auction_start_time: FixedType,
+}
+
+trait GdaTrait {
```
Comment (tarrencev) [inline crates/dojo-defi/src/dutch_auction/gda.cairo:12]:
```suggestion
trait GradualDutchAuctionTrait {
```

I would prefer the full name here

File: crates/dojo-defi/src/dutch_auction/gda.cairo [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    fn purchase_price(
+        self: @Market, quantity: u128, existing: u128, current_time: u128
+    ) -> FixedType;
+}
+
+impl GdaImpl of GdaTrait {
```
Comment (tarrencev) [inline crates/dojo-defi/src/dutch_auction/gda.cairo:18]:
```suggestion
impl GradualDutchAuctionImpl of GradualDutchAuctionTrait {
```

File: crates/dojo-defi/src/dutch_auction/vrgda.cairo [hunk 5]
```diff
@@ -0,0 +1,35 @@
+use cubit::types::fixed::{Fixed, FixedAdd, FixedMul, FixedSub, FixedDiv};
+
+#[derive(Component)]
+#[derive(Drop)]
+struct Market {
```
Comment (tarrencev) [inline crates/dojo-defi/src/dutch_auction/vrgda.cairo:5]:
```suggestion
struct VariableGradualDutchAuction {
```

File: crates/dojo-defi/Scarb.toml [hunk 6]
```diff
@@ -4,7 +4,7 @@ version = "0.1.0"
 description = "Implementations of a defi primitives for the Dojo framework"
 
 [dependencies]
-cubit = { version = "0.1.0", git = "https://github.com/influenceth/cubit" }
+cubit = { git = "https://github.com/influenceth/cubit", version = "0.1.0" }
 dojo_core = { path = "../dojo-core" }
 
-[[target.dojo]]
+# [[target.dojo]]
```
Comment (tarrencev) [inline crates/dojo-defi/Scarb.toml:10]: This is necessary to compile cairo that uses the `Component` macro

File: crates/dojo-defi/src/dutch_auction/vrgda.cairo [hunk 7]
```diff
@@ -0,0 +1,35 @@
+use cubit::types::fixed::{Fixed, FixedAdd, FixedMul, FixedSub, FixedDiv};
+
+#[derive(Component)]
```
Comment (ponderingdemocritus) [inline crates/dojo-defi/src/dutch_auction/vrgda.cairo:3]:
```suggestion
#[derive(Component, Drop)]
```

File: crates/dojo-defi/src/dutch_auction/vrgda.cairo [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+#[derive(Drop)]
```
Comment (ponderingdemocritus) [inline crates/dojo-defi/src/dutch_auction/vrgda.cairo:4]:
```suggestion
```

**Comment by ponderingdemocritus** [comment]
@ametel01 could you please fix conflicts and comments, then we can merge. It would be worth creating an example, but we can do that in another PR.

**Comment by tarrencev** [comment]
@ametel01 could you please add tests for these?

**Comment by ametel01** [comment]
> @ametel01 could you please add tests for these?

opened a draft PR #810

**Pull request merged**